### DISCUSSION TITLE
Relative merits of lgpio & RPi.GPIO factories?

### DISCUSSION THREAD

**Discussion by breaker-of-stone** [gpiozero/gpiozero#1077] [Q&A]
Hi.  Starting v0.2 of a community robotics project, and puzzled by the factory choices this time around.

We used a mix of Mock factory (for developing on Mac/PC) and lgpio under Ubuntu on R.Pi.4 for our first generation prototype.
We reasoned that:

- the switch to an ABI meant we had to get off older sysfs-style factories
- lgpio was the only ABI enabled factory that had I2C, SPI and PWM back in mid 2022
- gpiozero was defaulting its pin factory to lgpio, if it was available - signifiying its full feature capability

... and it all worked brilliantly.

In r0.2 we are switching to Manjaro on R.Pi.4 for a number of reasons, e.g. more mature Pipewire + Wireplumber because we need fancy audio.
However:
- lgpio is a pig to build on Manjaro/Arch, and supporting packages are rare or non-existent
- gpiozero, our core framework has switched its default factory back to RPi.GPIO

Can we confirm whether:
RPi.GPIO has now migrated to the ABI & away from sysfs
RPI.GPIO now provides full support via the ABI for I2C, SPI & PWM 

... and therefore we should stop banging our heads against lgpio builds on Manjaro?
The relief would be wonderful!

**Comment by bennuttall** [comment]
There is no single "default" pin factory, it's actually an ordered preference list: RPi.GPIO, lgpio, rpio, pigpio and native as per the docs: https://gpiozero.readthedocs.io/en/stable/api_pins.html#changing-the-pin-factory

If a pin factory is available, we use it. If it's not available, we try the next one. So in your case, if RPi.GPIO was not found, we'll use it. If it isn't, we'll try lgpio, and so on. However, if you have a preference, you can override the default with your preference, as long as it's installed and available. Again, see the pin factory docs.

I don't know anything about RPi.GPIO's plans (it's a separate project) but there's no mention of sysfs in its release notes (last release was over a year ago: https://pypi.org/project/RPi.GPIO/)

**Reply by breaker-of-stone** [reply]
Dave Jones was able to advise that RPi.GPIO continues to rely on sysfs, so we're stuck with lgpio if we need Python to operate via the ABI.  Bit of a nuisance, but we're minimising our exposure (thread in the comments):

https://waldorf.waveform.org.uk/2021/the-pins-they-are-a-changin.html#disqus_thread
